Team417_2020/src/main/java/org/firstinspires/ftc/team417_2020/AutoCompetition.java

```diff
@@ -9,67 +9,72 @@
 import org.openftc.easyopencv.OpenCvCameraRotation;
 
 @Autonomous(name="Competition Auto")
-public class AutoCompetition extends MasterAutonomous{
+public class AutoCompetition extends MasterAutonomous {
 
     int allianceSide = 1;
     Toggler rightBumper = new Toggler();
 
+    // Constants
+    static final int MINIMUM_RING_AREA = 1800;
+    static final int PARKING_Y_POSITION = 84;
+    static final int A_C_TARGET_ZONE_X_POSITION = -24;
+    static final int A_TARGET_ZONE_Y_POSITION = 72;
+    static final int B_TARGET_ZONE_Y_POSITION = 96;
+    static final int C_TARGET_ZONE_Y_POSITION = 120;
+
     RingDetectionOpenCV ringDetector = new RingDetectionOpenCV();
     OpenCvCamera webcam;
     int numRings;
 
-
-
     @Override
     public void runOpMode() throws InterruptedException {
         autoInitializeRobot();
         initializeRingCounter();
 
-
         telemetry.addLine("waiting for start");
-        telemetry.update();
 
         while (!opModeIsActive()) {
             countRings();
             pickAllianceSide();
             telemetry.update();
         }
 
-
         waitForStart();
 
         // 0 rings -> Zone A
         // 1 rings -> Zone B
         // 4 rings -> Zone C
-        if (numRings == 0) {
-            // Nagivate to Zone A
-            move(0, 72, 0.7);
-            move(-24, 72, 0.7);
-            openWobbleGoalArm();
-            move(-24, 48, 0.7);
-            move(0, 84, 0.7);
-
-        } else if (numRings == 1) {
-            // Navigate to Zone B
-            move(0, 96, 0.7);
-            openWobbleGoalArm();
-            move(0, 84, 0.7);
-
-        } else if (numRings == 4) {
-            // Navigate to Zone C
-            move(0, 120, 0.7);
-            move( -24, 120, 0.7);
-            openWobbleGoalArm();
-            move(0, 84, 0.7);
-
+        switch (numRings) {
+            case 0:
+                // Nagivate to Zone A
+                move(0, A_TARGET_ZONE_Y_POSITION, 0.7);
+                move(A_C_TARGET_ZONE_X_POSITION, A_TARGET_ZONE_Y_POSITION, 0.7);
+                openWobbleGoalArm();
+                move(A_C_TARGET_ZONE_X_POSITION, 48, 0.7);
+
```

Comment:
With a switch statement you have to break in each case otherwise it'll do everything in the other cases as well. This website explains how to use it correctly. https://www.w3schools.com/java/java_switch.asp